library/Translate/test/Test/Golden.hs

```diff
@@ -1,35 +1,315 @@
-import Juvix.Frontend.Types (TopLevel)
+import qualified Juvix.Frontend.Sexp as SexpTrans
+import Juvix.Frontend.Types (ModuleOpen (..), TopLevel)
 import Juvix.Frontend.Types.Base (Header)
 import Juvix.Library
+import qualified Juvix.Library.Feedback as Feedback
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import Juvix.Library.Parser (Parser)
+import qualified Juvix.Library.Parser as J
 import Juvix.Library.Test.Golden
+import qualified Juvix.Sexp as Sexp
+import qualified System.FilePath as FP
 import Test.Tasty
+import qualified Text.Megaparsec as P
+import qualified Text.Megaparsec.Byte as P
+import Prelude (String, error)
+
+juvixRootPath :: FilePath
+juvixRootPath = "../../"
+{-# INLINE juvixRootPath #-}
+
+withJuvixRootPath :: FilePath -> FilePath
+withJuvixRootPath p = juvixRootPath <> p
+
+withJuvixStdlibPath :: FilePath -> FilePath
+withJuvixStdlibPath p = juvixRootPath <> "stdlib/" <> p
+
+withJuvixExamplesPath :: FilePath -> FilePath
+withJuvixExamplesPath p = juvixRootPath <> "test/examples/" <> p
+{-# INLINE withJuvixExamplesPath #-}
+
+top :: IO TestTree
+top =
+  testGroup
+    "golden tests"
+    <$> sequence
+      [parseTests, desugartests, contextTests]
 
 --------------------------------------------------------------------------------
--- Parse contracts (Golden tests)
+-- Parse Test Frame
+--------------------------------------------------------------------------------
+
+positiveTests ::
+  (Applicative f) => TestName -> (FilePath -> f [TestTree]) -> f TestTree
+positiveTests testName discoverFunction =
+  testGroup testName
+    <$> sequenceA
+      [ testGroup "positive"
+          <$> discoverFunction (withJuvixExamplesPath "positive")
+      ]
+
+negativeTests ::
+  (Applicative f) => TestName -> (FilePath -> f [TestTree]) -> f TestTree
+negativeTests testName discoverFunction =
+  testGroup testName
+    <$> sequenceA
+      [ testGroup "negative"
+          <$> discoverFunction (withJuvixExamplesPath "negative")
+      ]
+
+--------------------------------------------------------------------------------
+-- Feedback
 --------------------------------------------------------------------------------
 
--- TODO: Add translations as well?
+type Feedback = Feedback.FeedbackT [] String IO
+
+--------------------------------------------------------------------------------
+-- Parse contracts (Golden tests)
+--------------------------------------------------------------------------------
 
 parseContract :: FilePath -> IO (Either [Char] (Header TopLevel))
 parseContract file = do
   Parser.prettyParse <$> ByteString.readFile file
 
 parseTests :: IO TestTree
 parseTests =
-  testGroup "parse"
-    <$> sequence
-      [ discoverGoldenTestsParse "../../test/examples/positive",
-        discoverGoldenTestsParse "../../test/examples/negative"
-      ]
+  positiveTests "parse" (fmap (: []) . discoverGoldenTestsParse)
+
+desugartests :: IO TestTree
+desugartests = positiveTests "desugar" discoverGoldenTestsDesugar
```

Comment:
Want to add negativeTests?

Reply:
negative tests for this is a bit rough, As at what phase do we expect to fail? We probably want some scaffolding to say when the test should fail